Library/Homebrew/cask/cask_loader.rb

```diff
@@ -110,9 +114,8 @@ def initialize(path, token: T.unsafe(nil))
         path = Pathname(path).expand_path
 
         @token = path.basename(path.extname).to_s
-
         @path = path
-        @tap = Homebrew::API.tap_from_source_download(path)
+        @tap = Tap.from_path(path) || Homebrew::API.tap_from_source_download(path)
```

Comment:
I think this might even be a touch faster here. The `Homebrew::API.tap_from_source_download` method is quite slow when it gets called a lot in commands like `brew readall`. This change in logic means it will get called less in that scenario though I plan on opening a PR soon to move from `Pathname#relative_path_from` in that method to a regex to speed things up anyway.